ocaml/alerts/expiry_alert.mli

```diff
@@ -0,0 +1,57 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+val update :
```

Comment:
These functions need a details explanation of what they do and why they tale all these parameters.

Avoiding complex functions is why I preferred using the functor, the client module does not need to know about complex functions, instead it provides information topt he functor so it can do its job.

Reply:
Add type definitions and comments to make it easy to understand in the header file: https://github.com/xapi-project/xen-api/pull/5060/files#diff-318524569501cd50cb93b08d9e5e86f2c11b3d61946f85b2ac38ea65ef68116e